steering_controllers/steering_odometry/include/steering_odometry/steering_odometry.hpp

```diff
@@ -0,0 +1,197 @@
+/*********************************************************************
+* Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ *********************************************************************/
+
+/*
+ * Author: dr. sc. Tomislav Petkovic
+ * Author: Dr. Ing. Denis Stogl
+ */
+
+#pragma once
+
+#include "realtime_tools/realtime_buffer.h"
+#include "realtime_tools/realtime_publisher.h"
+
+#include <boost/function.hpp>
+#include "rcpputils/rolling_mean_accumulator.hpp"
+
+namespace steering_odometry
+{
+/**
+   * \brief The Odometry class handles odometry readings
+   * (2D pose and velocity with related timestamp)
+   */
+class SteeringOdometry
+{
+public:
+  /// Integration function, used to integrate the odometry:
+  typedef boost::function<void(double, double)> IntegrationFunction;
```

Comment:
I am not sure if we really need this layer of abstraction?

Can we just use on all the time? Or are there two different cases?